draft-ietf-mls-protocol.md

```diff
@@ -1614,14 +1615,21 @@ struct {
 } MLSPlaintextTBS;
 ~~~~~
 
-<!-- OPEN ISSUE: group_id and epoch are duplicated in the TBS and in
-GroupContext. Think about how to de-duplicate. -->
+The `membership_token` field in the MLSPlaintext object authenticates the
+sender's membership in the group.  For an MLSPlaintext with a sender type other
+than `member`, this field MUST be set to the zero-length octet string.  For
+messages sent by members, it MUST be set to the following value:
 
-The ciphertext field of the MLSCiphertext object is produced by
-supplying the inputs described below to the AEAD function specified
-by the ciphersuite in use.  The plaintext input contains content and
-signature of the MLSPlaintext, plus optional padding.  These values
-are encoded in the following form:
+~~~~~
+membership_token = KDF.Extract(confirmation_key, MLSPlaintextTBS);
```

Comment:
Using a KDF output for authentication feels wrong

Reply:
We already do this elsewhere:
```
MLSPlaintext.confirmation_tag =
    KDF.Extract(confirmation_key, GroupContext.confirmed_transcript_hash)
```

This seems OK with HKDF, since Extract is just HMAC.  But I agree that we should confirm that this generalizes properly.

Reply:
Agreements on the 2020-08-25 call:
* Use an explicit MAC here instead of KDF.Extract
* Use a separate key from the key schedule, different from the confirmation_key

```
epoch
  |
  +-- confirmation_key
  |
  +-- membership_key
  |
  +-- authentication_secret (todo)
```